test/test_helper.rb

```diff
@@ -18,27 +18,31 @@
 
 require "kennel"
 
-class TestProject < Kennel::Models::Project
-  defaults(
-    team: -> { TestTeam.new },
-    parts: -> { [] }
-  )
-end
+Minitest::Test.class_eval do
+  def self.with_test_classes
```

Comment:
should this not clean up the classes after too ?
```
def with_test_classes
  around do |t|
    classes
    t.call
  ensure
    delete classes
  end
end
```

Reply:
also a bit sus that it calls when the test is loaded and not when the test is run
... might as well always create the classes in test_helper if we really need this, because otherwise we have different behavior depending on what test files are loaded

Reply:
The criticisms you make apply equally well to the pre-PR code too. In fact even more so, because the whole point of `with_test_classes` is to make that suspect behaviour optional.

Reply:
before everything had the same environment
after:
- the classes are defined multiple times
- depending on the load-order of the tests different classes are defined

Reply:
Repeating yourself can be fine. Can be a _good_ thing.

Could you be more specific about the repeat definitions / the load order thing?

Reply:
when multiple tests are loaded the method gets called multiple times
so the classes get re-defined multiple times (right ?)

when multiple tests get loaded at the same time the defined classes will be different from when only a single test is loaded
... so having a single global env might be simpler and easier to debug

Reply:
> when multiple tests are loaded the method gets called multiple times 

Does that ever happen? AFAICT each test file runs in its own process.

> when multiple tests get loaded at the same time

See above.

Reply:
ah true! :D
... yeah with forking-test-runner this should be fine 👍